src/dotnet-roslyn-tools/Commands/PRTaggerCommand.cs

```diff
@@ -49,12 +44,30 @@ public async Task<int> InvokeAsync(InvocationContext context)
                 return -1;
             }
 
+            var vsBuildNumber = context.ParseResult.GetValueForOption(maxVsBuildCheckNumber);
+            logger.LogInformation($"Check {vsBuildNumber} VS Builds");
+
+            using var devdivConnection = new AzDOConnection(settings.DevDivAzureDevOpsBaseUri, "DevDiv", settings.DevDivAzureDevOpsToken);
+            var buildsAndCommits = await PRTagger.PRTagger.GetVSBuildsAndCommitsAsync(devdivConnection, logger, vsBuildNumber, CancellationToken.None).ConfigureAwait(false);
+
+            var client = new HttpClient
+            {
+                BaseAddress = new("https://api.github.com/")
+            };
+
+            client.DefaultRequestHeaders.Accept.Add(
+                new MediaTypeWithQualityHeaderValue("application/vnd.github.v3+json"));
+            client.DefaultRequestHeaders.Add("User-Agent", "roslyn-tool-tagger");
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue(
+                "Bearer",
+                settings.GitHubToken);
+
             return await PRTagger.PRTagger.TagPRs(
-                vsBuild,
-                vsCommitSha,
+                vsBuildsAndCommitSha: buildsAndCommits,
                 settings,
-                logger,
-                CancellationToken.None).ConfigureAwait(false);
+                devdivConnection,
+                client,
+                logger).ConfigureAwait(false);
```

Comment:
I would move this bit to a new method in the PRTagger class and call into it here. The commands are intended to be light on logic other than option parsing.